src/main/webapp/app/service/firebase/firebase-gene-review-service.ts

```diff
@@ -79,136 +58,124 @@ export class FirebaseGeneReviewService {
       updateObject[`${pathFromGene}_uuid`] = uuid;
     }
 
-    return this.firebaseRepository.update(getFirebaseGenePath(isGermline, hugoSymbol), updateObject).then(() => {
-      if (isChangeReverted) {
-        this.firebaseRepository.delete(`${firebasePath}_review/lastReviewed`);
-      }
-      // Update Meta information
-      this.firebaseMetaService.updateGeneMetaContent(hugoSymbol, isGermline);
-      this.firebaseMetaService.updateGeneReviewUuid(hugoSymbol, uuid, !isChangeReverted, isGermline);
-    });
+    try {
+      await this.firebaseMetaService.updateMeta(hugoSymbol, uuid, !isChangeReverted, isGermline);
```

Comment:
Assuming we didn't come up with a solution for if one of these updates failed then? I'm fine with leaving this, but if you think it's really critical to check both you can look at `createGene` in `firebase-gene-service.ts`. I created a utility method `isPromiseOk` for dealing with this sort of scenario.

Reply:
I took a look at using multi-location updates (atomic operation), however it does not work well when the update depends on the data that is currently in firebase. For instance, when we accept a deletion, we need to know the current index. This use case is what `runTransaction` is used for, but in order to update both Meta and Gene collections, we need to provide the root path. `runTransaction` also reads in the data given the path, so in our case, we would be reading in our entire firebase database.

I am going to use SentryError to track errors for now.